src/adapter/src/catalog.rs

```diff
@@ -1633,9 +1643,26 @@ impl ConnCatalog<'_> {
         &*self.state
     }
 
+    /// Prevent planning from resolving item with the provided ID. Instead,
+    /// return an error as if the item did not exist.
+    ///
+    /// This feature is meant exclusively to permit re-planning statements
+    /// during update operations.
+    ///
+    /// # Panics
+    /// If the catalog's role ID is not [`MZ_SYSTEM_ROLE_ID`].
+    pub fn mark_id_unresolvable(&mut self, id: GlobalId) {
```

Comment:
nit: maybe rename to `mark_id_unresolvable_for_replanning(...)`? It's a pretty long name but makes it hard to misuse

Reply:
Sounds good to me.